### PR TITLE
[MSHARED-1152] - Migrate tests to JUnit5

### PR DESCRIPTION
**Pull request by iofit**
Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [X] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/MSHARED) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [X] Each commit in the pull request should have a meaningful subject line and body.
 - [X] Format the pull request title like `[MSHARED-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MSHARED-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [X] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [X] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [X] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

File: src/test/java/org/apache/maven/shared/invoker/MavenCommandLineBuilderTest.java [hunk 1]
```diff
@@ -19,57 +19,46 @@
...
-
-import static org.hamcrest.CoreMatchers.is;
-import static org.hamcrest.CoreMatchers.notNullValue;
-import static org.junit.Assert.assertArrayEquals;
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertTrue;
-import static org.junit.Assert.fail;
-import static org.junit.Assume.assumeThat;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (slawekjaranowski) [inline src/test/java/org/apache/maven/shared/invoker/MavenCommandLineBuilderTest.java:36]: Don't use wildcard imports.

File: src/test/java/org/apache/maven/shared/invoker/MavenCommandLineBuilderTest.java [hunk 2]
```diff
@@ -19,57 +19,46 @@
...
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Properties;
-import java.util.Set;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
```
Comment (slawekjaranowski) [inline src/test/java/org/apache/maven/shared/invoker/MavenCommandLineBuilderTest.java:34]: Incorrect order.

File: src/test/java/org/apache/maven/shared/invoker/MavenCommandLineBuilderTest.java [hunk 3]
```diff
@@ -109,7 +98,7 @@ public void testShouldFailToSetLocalRepoLocationFromRequestWhenItIsAFile()
     @Test
     public void testShouldSetLocalRepoLocationGlobally() throws IOException
     {
-        File lrd = temporaryFolder.newFolder( "workdir" ).getCanonicalFile();
+        File lrd = Files.createDirectory(temporaryFolder.resolve("workdir")).toFile().getCanonicalFile();
```
Comment (slawekjaranowski) [inline src/test/java/org/apache/maven/shared/invoker/MavenCommandLineBuilderTest.java:101]: Please follow: https://maven.apache.org/developers/conventions/code.html in whole file.

Should be:
```suggestion
        File lrd = Files.createDirectory( temporaryFolder.resolve( "workdir" ) ).toFile().getCanonicalFile();
```

missing spaces

File: src/test/java/org/apache/maven/shared/invoker/MavenCommandLineBuilderTest.java [hunk 4]
```diff
@@ -857,46 +846,46 @@ public void testShouldInsertActivatedProfiles()
     public void testMvnExecutableFromInvoker()
         throws Exception
     {
-        assumeThat( "Test only works when maven.home is set",
-            System.getProperty( "maven.home" ), is(notNullValue()));
+        assumingThat( Objects.nonNull(System.getProperty( "maven.home" )),
+                () -> System.out.println("Test only works when maven.home is set"));
```
Comment (slawekjaranowski) [inline src/test/java/org/apache/maven/shared/invoker/MavenCommandLineBuilderTest.java:850]: `assumingThat` doesn't break a test ...

https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/Assumptions.html#assumingThat(boolean,org.junit.jupiter.api.function.Executable)

**Review by slawekjaranowski — CHANGES_REQUESTED** [review]

File: src/test/java/org/apache/maven/shared/invoker/MavenCommandLineBuilderTest.java (hunk 4 above)
Comment (iofit) [reply src/test/java/org/apache/maven/shared/invoker/MavenCommandLineBuilderTest.java]: Hi, `assumeThat` doesn't either, it ignores the test, and `assumingThat` does the same, do we want the test to fail if the assumption is not met? If so, that would be an assertion instead and we would have to rewrite expectations. 
[https://junit.org/junit4/javadoc/4.12/org/junit/Assume.html#assumeThat(java.lang.String,%20T,%20org.hamcrest.Matcher)](https://junit.org/junit4/javadoc/4.12/org/junit/Assume.html#assumeThat(java.lang.String,%20T,%20org.hamcrest.Matcher))

Comment (slawekjaranowski) [reply src/test/java/org/apache/maven/shared/invoker/MavenCommandLineBuilderTest.java]: Ok, I think that whole test should be ignored.
- old `assumeThat` skip test if condition is not meet
- but `assumingThat` only skips execution of provided lambda not whole test

Comment (iofit) [reply src/test/java/org/apache/maven/shared/invoker/MavenCommandLineBuilderTest.java]: I think you're right, I missed that specific part, thank you for catching it, I've replaced assumingThat with assumeTrue which should abort/skip the test if the condition is not met.

**Comment by jira-importer** [comment]
Resolve #148

**Pull request merged**